### PR TITLE
Add anonymous telemetry reporting.

### PR DESCRIPTION
**Pull request by danepowell**
Obviously lots of work still to do here.

- [x] Write a [systeminformation class](https://github.com/sebhildebrandt/systeminformation) that essentially replicates the Node.js version of same so we can get things like device ID and info in a unified way
- [ ] Create a very clear and explicit opt-in or opt-out system for telemetry reporting.

Right now this only logs invoked command names, which is a huge step forward but obviously not everything we'd like to collect. Long-term it would be nice to understand how often other features such as config management strategies, DrupalVM, Lando, etc... are being used.

### CODE REVIEW

File: src/Robo/Blt.php [hunk 1]
```diff
@@ -88,6 +89,23 @@ public function __construct(
...
+   */
+  private function initializeAmplitude() {
+    $amplitude = Amplitude::getInstance();
+    $amplitude->init('dfd3cba7fa72065cde9edc2ca22d0f37')
+      // @todo also set platform, arch, os_name, os_version, device_id
+      ->setUserProperties([
+        'blt-version' => $this::VERSION,
+      ])
+      // @todo get this from systeminformation
+      ->setDeviceId('foo')
```
[the remark this answers is not in the document]
Comment (danepowell) [reply src/Robo/Blt.php]: Yeah that was my point about having to write a systeminformation class. I'm going to get a unique machine id for each installation. So it'll be tracking devices / installs, not users.

**Comment by alexxed** [comment]
The oldest phpsysinfo I know is https://packagist.org/packages/phpsysinfo/phpsysinfo, but it has a different purpose, not sure if you can use it as a library and it's quite heavy, you probably need only a few of the things it offers.

For machine id, I don't know something in PHP, but what this project does https://github.com/denisbrodbeck/machineid is right and can be easily replicated in PHP.

**Comment by adrianthedev** [comment]
Couldn't we do the same thing we do with ads? Even more so, why don't we use the ads user config? `~/.config/ads-cli/user.json`. We could make it more of an "Acquia" specific config. If the user initialized ads, then ads writes the config, if not, blt writes this config. Make it interchangeable.

This way we have a more accurate way of tracking users across tools they use.

**Comment by danepowell** [comment]
I like the idea of sharing a single telemetry config / id across all Acquia products. Could we move it to `~/.config/acquia/user.json`? We could also dump API credentials in that directory, since right now every product stores them in its own way.

But we'd still need a way to generate the user info (system uuid, OS name, etc...) consistently across the products. To be consistent with ADS, I think that means literally porting https://github.com/sebhildebrandt/systeminformation to PHP.

**Comment by adrianthedev** [comment]
I found [this](http://phpsysinfo.github.io/phpsysinfo/) package but it doesn't seem to have it all. No machine UUID.

However, I found [this](https://github.com/sebhildebrandt/systeminformation/blob/master/lib/osinfo.js) file that gets the uuid.

**Comment by alexxed** [comment]
Using a single file across multiple products to determine if telemetry is sent or not is not correct to the user. He'd be prompted once and then we'd forward for other products.

Using the same algorithm to determine the device id makes sense, but we don't need to store it in the same file across products.

I'd recommend starting with device id, blt version, command that is being run, and the OS.

**Comment by danepowell** [comment]
I got the machine UUID and other system info generated consistently, now I just need to figure out how to attach it to Amplitude events. The zumba/amplitude-php maintainer made some suggestions here that I don't fully understand yet: https://github.com/zumba/amplitude-php/issues/17#issuecomment-495409923

**Comment by danepowell** [comment]
This is good to go functionally, just waiting for guidance on the opt-in/out and messaging requirements.

**Comment by danepowell** [comment]
I have no idea how long it's going to take to come up with the guidance on opt-in/opt-out and messaging, so I'm going to close this in the meantime to keep the queue clear.

**Pull request closed without merging**